### PR TITLE
fix: quota memory parsing

### PR DESCRIPTION
**Pull request by olevski**
Fixes this problem:

```
[2023-10-25 11:38:53 +0000] [17] [ERROR] An unknown or unhandled exception occurred
Traceback (most recent call last):
  File "handle_request", line 97, in handle_request
  File "/app/env/lib/python3.11/site-packages/renku_data_services/base_api/auth.py", line 25, in decorated_function
    response = await f(request, *args, **kwargs, user=user)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.11/site-packages/renku_data_services/crc_blueprints/core.py", line 35, in _get_all
    rps_w_quota = [self.quota_repo.hydrate_resource_pool_quota(rp) for rp in rps]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.11/site-packages/renku_data_services/crc_blueprints/core.py", line 35, in <listcomp>
    rps_w_quota = [self.quota_repo.hydrate_resource_pool_quota(rp) for rp in rps]
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.11/site-packages/renku_data_services/k8s/quota.py", line 131, in hydrate_resource_pool_quota
    quota = self._get_quota(resource_pool.quota)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.11/site-packages/renku_data_services/k8s/quota.py", line 69, in _get_quota
    return self._quota_from_manifest(res_quota)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.11/site-packages/renku_data_services/k8s/quota.py", line 37, in _quota_from_manifest
    memory=round(ByteSize(memory_raw).to("G")),
                 ^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '250G'
```

Luckily Wes noticed this on dev.renku.ch before it is rolled out everywhere.

### CODE REVIEW

File: components/renku_data_services/k8s/quota.py [hunk 1]
```diff
@@ -30,11 +41,12 @@ def _quota_from_manifest(self, manifest: client.V1ResourceQuota) -> models.Quota
                 gpu_kind = igpu_kind
                 break
         memory_raw = manifest.spec.hard.get("requests.memory")
-        if memory_raw[-1] == "i":
+        # NOTE: Pydantic will fail to parse "Gi" but will be happy with "Gib", but Kubernetes works with Gi or G
```
Comment (leafty) [inline components/renku_data_services/k8s/quota.py:44]: Would it not be safer to use the `parse_quantity()` util from the `kubernetes` library?

Comment (olevski) [reply components/renku_data_services/k8s/quota.py:44]: I did not know about this. That is probably better.

Comment (leafty) [reply components/renku_data_services/k8s/quota.py:44]: Technically, all fields from `manifest.spec.hard` are quantities, so `cpu` may also be specified as `2500m` (2.5).

Comment (olevski) [reply components/renku_data_services/k8s/quota.py]: Done.

**Review by mohammad-alisafaee — APPROVED** [review]
LGTM, thank you!

**Pull request merged**